sklearn/ensemble/bagging.py

```diff
@@ -387,6 +397,58 @@ def _validate_y(self, y):
         # Default implementation
         return column_or_1d(y, warn=True)
 
+    def _get_estimators_data_draws(self, sample_mask=False, feature_mask=False):
+        # If max_samples is float
+        if not isinstance(self.max_samples, (numbers.Integral, np.integer)):
+            max_samples = int(self.max_samples * self._n_samples)
+
+        if not (0 < max_samples <= self._n_samples):
+            raise ValueError("max_samples must be in (0, n_samples]")
+
+        # If max_features is float
+        if not isinstance(self.max_features, (numbers.Integral, np.integer)):
+            max_features = int(self.max_features * self.n_features_)
+
+        if not (0 < max_features <= self.n_features_):
+            raise ValueError("max_features must be in (0, n_samples]")
+
+        # Get sampled indices or mask
+        for seed in self._seeds:
+            # Operations accessing random_state must be performed identically
+            # to those in `_parallel_build_estimators()`
+            random_state = check_random_state(seed)
+            seed = random_state.randint(MAX_INT) 
+            feature_indices, sample_indices = _generate_bagging_indices(random_state, 
+                                                                  self.bootstrap_features,
+                                                                  self.bootstrap, 
+                                                                  self.n_features_, 
+                                                                  self._n_samples, 
+                                                                  max_features, 
+                                                                  max_samples)
+
+            # Convert feature indices to mask if desired
+            if feature_mask:
+                feature_draw = np.zeros(self._n_samples, dtype=bool)
+                feature_draw[feature_indices] = True
+            else:
+                feature_draw = feature_indices
+
+            # Convert sample indices to mask if desired
+            if sample_mask:
+                sample_draw = np.zeros(self._n_samples, dtype=bool)
+                sample_draw[sample_indices] = True
+            else:
+                sample_draw = sample_indices
+
+            yield feature_draw, sample_draw
+
+    @property
+    @deprecated("Attribute 'estimator_samples_' is deprecated and will be removed "
+                "in release 0.20. Use method '_get_estimators_data_draw()' instead.")
```

Comment:
We can't advise to use a private method. If we need to continue to provide users with access, we can not deprecate this (just regenerate it at every access), or we can provide a generator as a new attribute/method.